test/extended/authentication.sh

```diff
@@ -65,28 +65,65 @@ wait_for_command 'oc get imagestream openldap --template="{{with \$tags := .stat
 
-
 oc login -u system:admin -n default
 
+# readiness check is somehow faulty
+sleep 10
+
 echo "[INFO] Running extended tests"
 
-sleep 10
+# create a temp copy of the sync-configs and update them with the LDAP server's IP
+for config in test/extended/authentication/sync-config*.yaml
+do
+	cp ${config} ${BASETMPDIR}
+	os::util::sed "s/LDAP_SERVICE_IP/${LDAP_SERVICE_IP}/g" ${BASETMPDIR}/$(basename ${config})
+done
+
+# sync all groups from LDAP using rfc2307 schema, expect failure to find Person0
+! openshift ex sync-groups --sync-config=${BASETMPDIR}/sync-config-rfc2307.yaml --confirm 
+touch test/extended/authentication/valid_all_ldap_rfc2307_sync.txt
+oc get groups --no-headers | awk '{print $1}' | xargs -I{} oc get group {} -o yaml >> test/extended/authentication/valid_all_ldap_rfc2307_sync.txt
+oc delete groups --all
 
-cp test/extended/authentication/sync-schema1-group1.yaml ${BASETMPDIR}
-os::util::sed "s/LDAP_SERVICE_IP/${LDAP_SERVICE_IP}/g" ${BASETMPDIR}/sync-schema1-group1.yaml
-openshift ex sync-groups --sync-config=${BASETMPDIR}/sync-schema1-group1.yaml --confirm
+# sync all groups from LDAP using ad schema, 
+openshift ex sync-groups --sync-config=${BASETMPDIR}/sync-config-ad.yaml --confirm
+touch test/extended/authentication/valid_all_ldap_ad_sync.txt
+oc get groups --no-headers | awk '{print $1}' | xargs -I{} oc get group {} -o yaml >> test/extended/authentication/valid_all_ldap_ad_sync.txt
+oc delete groups --all
 
+# sync some groups from whitelist file from LDAP server with rfc2307 schema
+openshift ex sync-groups --whitelist=test/extended/authentication/whitelist_ldap.txt --sync-config=${BASETMPDIR}/sync-config-rfc2307.yaml --confirm
+touch test/extended/authentication/valid_whitelist_ldap_rfc2307_sync.txt
+oc get groups --no-headers | awk '{print $1}' | xargs -I{} oc get group {} -o yaml >> test/extended/authentication/valid_whitelist_ldap_2307_sync.txt
 oc delete groups --all
 
-cp test/extended/authentication/sync-schema2-01.yaml ${BASETMPDIR}
-os::util::sed "s/LDAP_SERVICE_IP/${LDAP_SERVICE_IP}/g" ${BASETMPDIR}/sync-schema2-01.yaml
-openshift ex sync-groups --sync-config=${BASETMPDIR}/sync-schema2-01.yaml --confirm
+# sync some groups from whitelist from LDAP server with rfc2307 schema
+openshift ex sync-groups cn=group1,ou=groups,dc=example,dc=com --sync-config=${BASETMPDIR}/sync-config-rfc2307.yaml --confirm
+touch test/extended/authentication/valid_whitelist_ldap_rfc2307_sync2.txt
+oc get groups --no-headers | awk '{print $1}' | xargs -I{} oc get group {} -o yaml >> test/extended/authentication/valid_whitelist_ldap_2307_sync2.txt
+oc delete groups --all
 
+# sync group from LDAP
+openshift ex sync-groups cn=group2,ou=groups,dc=example,dc=com --sync-config=${BASETMPDIR}/sync-config-rfc2307.yaml --confirm
+# add new member to the group in LDAP
+ldapmodify -x -h ${LDAP_SERVICE_IP} -p 389 -D cn=Manager,dc=example,dc=com -w admin -f test/extended/authentication/modify_group1.ldif
```

Comment:
We can do both, wasn't the entire point of installing LDAP client tools on the Jenkins ami to enable this?